R/db.R

```diff
@@ -297,6 +297,14 @@ write_data_safely <- function(dbconn, tblname, df, key, safe_columns) {
   }
 
   write_column_hashed_data <- function(df, tblname, append = TRUE) {
+    ## Don't cache anything that is in the blacklist.
+    if (!is.null(blacklist)) {
+      if (any(is.na(blacklist))) {
+        df <- df[apply(df, 1, function(x) all(!(is.na(x)))),]
+      }
+      df <- df[apply(df, 1, function(x) all(!(x %in% blacklist))),]
+    }
+
```

Comment:
Correct me if I'm wrong, but AFAIK this will work like this
```r
# calling first time
do.call(cached_fn, params)
#   id name last_name returning
# 1 42 John     Smith     FALSE
#
# 10 weeks later...
# retrieving result from cache
do.call(cached_fn, params)
#   id name last_name
# 1 42 John     Smith
```